src/main/java/org/xerial/snappy/SnappyInputStream.java

```diff
@@ -93,20 +92,26 @@ protected void readHeader() throws IOException {
             return;
         }
 
+        if(!isValidHeader(header)) {
```

Comment:
oh really? I'll add a test to read the result of Snappy.compress with SnappyInputStream.